src/pico/i_video.c

```diff
@@ -949,108 +797,102 @@ void __noinline new_frame_init_overlays_palette_and_wipe() {
 // this method moved out of scratchx because we didn't have quite enough space for core1 stack
 void __no_inline_not_in_flash_func(new_frame_stuff)() {
     // this part of the per frame code is in RAM as it is needed during save
-    if (sem_available(&render_frame_ready)) {
-        sem_acquire_blocking(&render_frame_ready);
-        display_video_type = next_video_type;
-        display_frame_index = next_frame_index;
-        display_overlay_index = next_overlay_index;
-#if !DEMO1_ONLY
-        video_scroll = next_video_scroll; // todo does this waste too much space
-#endif
-        sem_release(&display_frame_freed);
-    } else {
+    sem_acquire_blocking(&render_frame_ready);
```

Comment:
This was the race condition that was causing the display to freeze if it was too fast. I don't entirely understand what the problem was but the change here is to _always_ wait for a render frame. When we were rendering VGA, there may have been a circumstance where the next frame isn't ready but our VGA signal needs a frame _now_ - in which case, we wouldn't advance the frame marker and swap to a new buffer, we'd just display the same frame again. This cannot happen to use now, since we can transmit new frames whenever we like, so changing this code to always block until the render frame is ready is correct. It also happened to fix the race condition, so that's swell